libs/s25main/ingameWindows/iwStatistics.cpp

```diff
@@ -301,6 +305,37 @@ void iwStatistics::DrawPlayerBoxes()
     }
 }
 
+void iwStatistics::DrawPlayerAlliances()
+{
+    constexpr unsigned spacingAllianceRects = 1;
+    constexpr Extent allianceRectExtent(4, 6);
+    constexpr unsigned allianceRectOffset = allianceRectExtent.x + spacingAllianceRects;
+
+    const GameWorldBase& world = gwv.GetWorld();
+    unsigned short startX = 126 - numPlayingPlayers * 17 + spacingAllianceRects;
+    DrawPoint drawPt = GetDrawPos() + DrawPoint(startX, 22 + spacingAllianceRects);
+    for(auto idx : helpers::range(world.GetNumPlayers()))
+    {
+        const GamePlayer& curPlayer = world.GetPlayer(idx);
+        if(!curPlayer.isUsed())
+            continue;
+
+        unsigned pactPos = 0;
+        for(auto idx_other : helpers::range(world.GetNumPlayers()))
+        {
+            const GamePlayer& otherPlayer = world.GetPlayer(idx_other);
+            if(!curPlayer.IsAlly(idx_other) || idx_other == idx)
+                continue;
+
+            auto const playerAllianceBoxRect =
+              Rect(drawPt + DrawPoint(pactPos * (allianceRectOffset), 0), allianceRectExtent);
```

Comment:
```suggestion
              Rect(drawPt + DrawPoint(pactPos * allianceRectOffset, 0), allianceRectExtent);
```

Or better yet: Initialize `pactPos = drawPt` and use `pactPos.x += allianceRectOffset`